client/src/tasks/msp_remove_finalised_files.rs

```diff
@@ -0,0 +1,136 @@
+    Runtime: StorageEnableRuntime,
+{
+    pub fn new(storage_hub_handler: StorageHubHandler<NT, Runtime>) -> Self {
+        Self {
+            storage_hub_handler,
+        }
+    }
+
+    async fn remove_file_from_file_storage(&self, file_key: &sp_core::H256) -> anyhow::Result<()> {
+        // Remove the file from the File Storage.
+        let mut write_file_storage = self.storage_hub_handler.file_storage.write().await;
+        write_file_storage.delete_file(file_key).map_err(|e| {
+					error!(target: LOG_TARGET, "Failed to remove file from File Storage after it was removed from the Bucket's Forest. \\nError: {:?}", e);
+					anyhow!(
+							"Failed to delete file from File Storage after it was removed from the Bucket's Forest: {:?}",
+							e
+					)
+			})?;
+        // Release the file storage write lock.
+        drop(write_file_storage);
+
+        Ok(())
+    }
+}
+
+/// Handles the [`FinalisedBucketMutationsApplied`] event.
+impl<NT, Runtime> EventHandler<FinalisedBucketMutationsApplied<Runtime>>
+    for MspRemoveFinalisedFilesTask<NT, Runtime>
+where
+    NT: ShNodeType<Runtime> + 'static,
+    NT::FSH: MspForestStorageHandlerT<Runtime>,
+    Runtime: StorageEnableRuntime,
+{
+    async fn handle_event(
+        &mut self,
+        event: FinalisedBucketMutationsApplied<Runtime>,
+    ) -> anyhow::Result<()> {
+        info!(
+            target: LOG_TARGET,
+            "Processing finalised bucket mutations applied for bucket [{:?}] with mutations: {:?}",
+            event.bucket_id,
+            event.mutations
+        );
+
+        // For each mutation...
+        for mutation in event.mutations {
+            let file_key = FileKey::from(mutation.0);
+
+            // Check that the file_key is not in the Bucket's Forest.
+            let bucket_forest_key = event.bucket_id.as_ref().to_vec();
+            let read_fs = self
+                .storage_hub_handler
+                .forest_storage_handler
+                .get(&bucket_forest_key)
+                .await
+                .ok_or_else(|| {
+                    anyhow!(
+                        "CRITICAL❗️❗️ Failed to get forest storage for bucket [{:?}].",
+                        event.bucket_id
+                    )
+                })?;
+            if read_fs.read().await.contains_file_key(&file_key.into())? {
+                warn!(
+                    target: LOG_TARGET,
+                    "BucketMutationsApplied and finalised for file key {:?} in bucket {:?}, but file key is still in Forest. This can only happen if the same file key was added again after deleted by the user.\\n Mutation: {:?}",
+                    file_key,
+                    event.bucket_id,
+                    mutation
+                );
+            } else {
+                // If file key is not in Forest, we can now safely remove it from the File Storage.
+                self.remove_file_from_file_storage(&file_key.into()).await?;
+            }
```

Comment:
We should be explicit about the type of Mutation this event handler is expecting since it is intended solely for deleting. Only delete from file storage if the mutation was a `Remove`.

https://github.com/Moonsong-Labs/storage-hub/blob/cb5084c03d7f8e9fdeea737f9f3623822702a535/primitives/traits/src/lib.rs#L928-L933